### PR TITLE
feat(dpg): rename `top` parameter in pagination method

### PR DESCRIPTION
**Pull request by archerzz**
- rename `top` parameter to `maxCount` if the name is not occupied
- add test

part of https://github.com/Azure/azure-sdk-for-net/issues/29342

# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/LowLevel/Output/DpgOutputLibraryBuilder.cs [hunk 1]
```diff
@@ -133,6 +133,47 @@ private static ClientInfo CreateClientInfo(InputClient ns, SourceInputModel? sou
             return new ClientInfo(ns.Name, clientName, clientNamespace, clientDescription, existingType, operations, clientParameters, resourceParameters);
         }
 
+        private static IReadOnlyList<InputOperation> UpdateOperations(IReadOnlyList<InputOperation> originalOperations)
```
Comment (AlexanderSher) [inline src/AutoRest.CSharp/LowLevel/Output/DpgOutputLibraryBuilder.cs:136]: If this change is applied to `InputOperation`, we should be able to call this at the beginning of `Build` method, after or before `UpdateListMethodNames`.

File: src/AutoRest.CSharp/LowLevel/Output/DpgOutputLibraryBuilder.cs [hunk 2]
```diff
@@ -133,6 +133,47 @@ private static ClientInfo CreateClientInfo(InputClient ns, SourceInputModel? sou
...
+            {
+                return operationParameters;
+            }
+
+            var parameters = new List<InputParameter>(operationParameters.Count);
+            foreach (var parameter in operationParameters)
+            {
+                if (parameter.Name.Equals("top", StringComparison.OrdinalIgnoreCase))
+                {
+                    parameters.Add(parameter with { Name = "maxCount" });
```
Comment (AlexanderSher) [inline src/AutoRest.CSharp/LowLevel/Output/DpgOutputLibraryBuilder.cs:166]: can we store `maxCount` as constant somewhere?

File: src/AutoRest.CSharp/LowLevel/Output/DpgOutputLibraryBuilder.cs [hunk 3]
```diff
@@ -133,6 +133,47 @@ private static ClientInfo CreateClientInfo(InputClient ns, SourceInputModel? sou
             return new ClientInfo(ns.Name, clientName, clientNamespace, clientDescription, existingType, operations, clientParameters, resourceParameters);
         }
 
+        private static IReadOnlyList<InputOperation> UpdateOperations(IReadOnlyList<InputOperation> originalOperations)
+        {
+            var operations = new List<InputOperation>(originalOperations.Count);
+            foreach (var operation in originalOperations)
+            {
+                if (operation.Paging != null)
```
Comment (AlexanderSher) [inline src/AutoRest.CSharp/LowLevel/Output/DpgOutputLibraryBuilder.cs:141]: What if we check for existing `maxCount` parameter right here? E.g.:
```cs
if (operation.Paging != null && !operation.Parameters.Any(p => p.Name.Equals("maxCount", StringComparison.OrdinalIgnoreCase)))
```

File: test/AutoRest.TestServerLowLevel.Tests/LowLevel/PaginationParameterOverwriteTests.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+using PaginationParams_LowLevel;
+
+namespace AutoRest.TestServer.Tests
+{
+    public class PaginationParameterOverwriteTests
+    {
+        [Test]
+        public void OverwriteTop()
+        {
+            Assert.AreEqual(new string[] { "maxCount", "skip", "maxpagesize", "context" }, GetMethodParameterNames("GetPaginationParamsAsync"));
```
Comment (AlexanderSher) [inline test/AutoRest.TestServerLowLevel.Tests/LowLevel/PaginationParameterOverwriteTests.cs:17]: Please use `TypeAsserts.HasPublicInstanceMethod`

File: src/AutoRest.CSharp/LowLevel/Output/DpgOutputLibraryBuilder.cs (hunk 3 above)
Comment (archerzz) [reply src/AutoRest.CSharp/LowLevel/Output/DpgOutputLibraryBuilder.cs:141]: @AlexanderSher That's correct.

**Comment by archerzz** [comment]
@AlexanderSher Thanks! Review comments are resolved. Could you please take a look again?

- move operation update into `UpdateListMethodNames` and rename it as `UpdateOperations`
- extract `maxCount` into a const string
- optimize the condition check for updating operation parameters
- add method assertion in test
- add missing test project launch setting

**Comment by lirenhe** [comment]
@archerzz, could you do a test run for azure-sdk-for-net?  
As mgmt. SDK would not apply this, we need to make sure there is no regression for them.

**Review by AlexanderSher — APPROVED** [review]

**Comment by archerzz** [comment]
> @archerzz, could you do a test run for azure-sdk-for-net? As mgmt. SDK would not apply this, we need to make sure there is no regression for them.

@lirenhe It's created: https://github.com/Azure/azure-sdk-for-net/pull/31220 Looks fine, however right now there is an issue blocking us from merging the PR.

**Comment by lirenhe** [comment]
> > @archerzz, could you do a test run for azure-sdk-for-net? As mgmt. SDK would not apply this, we need to make sure there is no regression for them.
> 
> @lirenhe It's created: [Azure/azure-sdk-for-net#31220](https://github.com/Azure/azure-sdk-for-net/pull/31220) Looks fine, however right now there is an issue blocking us from merging the PR.

@archerzz, could you check the comments from Heaths?  I suggest to have a codegen config to suppress rename for the existing services to avoid breaking change.

**Comment by archerzz** [comment]
> > > @archerzz, could you do a test run for azure-sdk-for-net? As mgmt. SDK would not apply this, we need to make sure there is no regression for them.
> > 
> > 
> > @lirenhe It's created: [Azure/azure-sdk-for-net#31220](https://github.com/Azure/azure-sdk-for-net/pull/31220) Looks fine, however right now there is an issue blocking us from merging the PR.
> 
> @archerzz, could you check the comments from Heaths? I suggest to have a codegen config to suppress rename for the existing services to avoid breaking change.

Per discussion, I've added a configuration `disable-pagination-top-renaming`. See the latest change here and in the preview SDK PR. I know this configuration name seems too fine-grained, but considering there are other pending issues in https://github.com/Azure/azure-sdk-for-net/issues/29342, I cannot see a better option.

**Comment by archerzz** [comment]
> > > > @archerzz, could you do a test run for azure-sdk-for-net? As mgmt. SDK would not apply this, we need to make sure there is no regression for them.
> > > 
> > > 
> > > @lirenhe It's created: [Azure/azure-sdk-for-net#31220](https://github.com/Azure/azure-sdk-for-net/pull/31220) Looks fine, however right now there is an issue blocking us from merging the PR.
> > 
> > 
> > @archerzz, could you check the comments from Heaths? I suggest to have a codegen config to suppress rename for the existing services to avoid breaking change.
> 
> Per discussion, I've added a configuration `disable-pagination-top-renaming`. See the latest change here and in the preview SDK PR. I know this configuration name seems too fine-grained, but considering there are other pending issues in [Azure/azure-sdk-for-net#29342](https://github.com/Azure/azure-sdk-for-net/issues/29342), I cannot see a better option.

Looks like this config is not required. Heaths has clarified that the SDK is not GAed.

**Pull request merged**